From c1566a1301533f3947e5c551479a256ba8c1e8eb Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 14 Aug 2018 17:01:00 +0300
Subject: [PATCH] cleanup: remove scratchpad 28 usage

Currently, only preboot is using the scratchpad 28.
It uses the scratchpad to pass the SPL offset value from one function
to another.
This use case is not justified enough to reserve a scratch pad.
Remove the use of scratchpad 28 - use variable (stack) instead.

Change-Id: I57818f2d878a3b9435a00396af4a1113e5107dcf
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/preboot.c b/preboot.c
index 4cb8104..cd02184 100644
--- a/preboot.c
+++ b/preboot.c
@@ -19,7 +19,7 @@
  * passing flash layout to CPU SW
  * Note: keep images size set by BTL
  */
-static void dali_preboot_copy_prepare(void)
+static void dali_preboot_copy_prepare(unsigned int *spl_off)
 {
 	unsigned int spl_offset = 0xa0000;
 	unsigned int uboot_offset = 0xb0000;
@@ -53,7 +53,7 @@
 	else /* Secondary */
 		image_base_offset = 0x800000;
 
-	writel((image_base_offset + spl_offset), mmPSOC_GLOBAL_CONF_SCRATCHPAD_28);
+	*spl_off = image_base_offset + spl_offset;
 	writel((image_base_offset + uboot_offset), mmPSOC_GLOBAL_CONF_SCRATCHPAD_30);
 }
 
@@ -62,8 +62,7 @@
 	void *spl_src, *spl_dst;
 	unsigned int i, spl_offset, spl_size;
 
-	dali_preboot_copy_prepare();
-	spl_offset = readl(mmPSOC_GLOBAL_CONF_SCRATCHPAD_28);
+	dali_preboot_copy_prepare(&spl_offset);
 	spl_src = (void*)HL_SPI_BASE + spl_offset;
 	spl_size = readl(mmPSOC_GLOBAL_CONF_SCRATCHPAD_29);
 	spl_dst = (void*)CONFIG_SPL_TEXT_BASE;
